src/index.js

```diff
@@ -1,49 +1,61 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { ThemeProvider } from 'react-jss';
+import DataContainer from 'elements/data-container';
+import RouteContainer from 'elements/route-container';
 import MainContainer from 'main-container';
+import state from 'state';
+import routes from 'routes';
+import css from 'css';
 
-global.React = React;
-
-// Prepend polyfill
-// from: https://github.com/jserz/js_piece/blob/master/DOM/ParentNode/prepend()/prepend().md
-(function(arr) {
-  arr.forEach(function(item) {
-    if (item.hasOwnProperty('prepend')) {
-      return;
-    }
-    Object.defineProperty(item, 'prepend', {
-      configurable: true,
-      enumerable: true,
-      writable: true,
-      value: function prepend() {
-        let argArr = Array.prototype.slice.call(arguments),
-          docFrag = document.createDocumentFragment();
-
-        argArr.forEach(function(argItem) {
-          let isNode = argItem instanceof Node;
-          docFrag.appendChild(
-            isNode ? argItem : document.createTextNode(String(argItem))
-          );
-        });
+require('utils/polyfills');
 
-        this.insertBefore(docFrag, this.firstChild);
-      },
-    });
-  });
-})([Element.prototype, Document.prototype, DocumentFragment.prototype]);
+global.React = React;
 
 const container = document.createElement('div');
 document.body.prepend(container);
 
 let Main = (global.Main = {});
 
-let main_props = {
-  main: Main,
-};
-Main.render = function() {
-  ReactDOM.render(React.createElement(MainContainer, main_props), container);
+const App = props => {
+  return (
+    <DataContainer
+      url="server/current-account"
+      onRequestComplete={async data => {
+        if (data.email) {
+          console.log(`[AUTH] Active User: ${data.email}`);
+          state.setActiveUser(data.email);
+          state.sync();
+        } else {
+          console.log(`[AUTH] Not logged in`);
+        }
+      }}
```

Comment:
Whether or not a user is logged in is the first question we ask upon loading the page, since that changes what is rendered, and what permissions a user has.